### DISCUSSION TITLE
[BUG] Invoke-PnPBatch : The maximum number of bytes allowed to be read from the stream has been exceeded. After the last read operation, a total of 1064000 bytes has been read from the stream; however a maximum of 1048576 bytes is allowed

### DISCUSSION THREAD

**Discussion by jolscr** [pnp/powershell#3466] [General]
### Reporting an Issue
I have a SharePoint script inventory that runs some cmdlet like "Set-PnPListItem" and "Add-PnPListItem" with bach parameter.
Then every 50 cmdlets, I run "Invoke-PnPBatch -Batch $batch -Connection $ConnectionSiteInventory"

When executing "Invoke-PnPBatch", randomly I receive the error below :
**Le nombre maximum d'octets pouvant être lus à partir du flux a été dépassé. Après la dernière opération de lecture, un total de 1064000 octets a été lu à partir du flux alors qu'un maximum de 1048576 octets est autorisé.**

In English I think it's: **The maximum number of bytes allowed to be read from the stream has been exceeded. After the last read operation, a total of 1064000 bytes has been read from the stream; however a maximum of 1048576 bytes is allowed**

Here is some statistic for 4 complete execution of my script:
- 77 Invoke-PnPBatch, I get 3 errors
- 30 Invoke-PnPBatch, I get 6 errors
- 35 Invoke-PnPBatch, I get 5 errors
- 73 Invoke-PnPBatch, I get 6 errors

Here is the SharePoint item property that I set:
![image](https://github.com/pnp/powershell/assets/46568866/f38d0ac8-047d-4ab9-9525-3ecc607dec63)

### Actual behavior 
![image](https://github.com/pnp/powershell/assets/46568866/2205bc23-f2fa-4419-a273-69acd8569895)

### What is the version of the Cmdlet module you are running?
![image](https://github.com/pnp/powershell/assets/46568866/582f65e9-0753-48c8-9a9e-2e0c61560ceb)

### Which operating system/environment are you running PnP PowerShell on?
- [x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
Am moving this to discussion since this is specific to your code/script. 
Would suggest maybe reducing the number of items to be sent in the batch and then try.

If anyone in the community has a suggestion, would be great to hear from them as well.

**Comment by jolscr** [comment]
Hello, 
I change my script to run 1 batch per request...
Now, I don't receive any error between sathurday and sunday, but monday I receive three errors when invoking pnp batch : 
**The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.**  (4:00, 4:02, 4:04) maybe network error because same hour...

My first issue is not resolve because I can't run multiple request with PnP Batch. If  anyone have same error message when executing PnPBatch with multiple column and value, please keep me informed.
